vadl/main/resources/templates/common/vadl-builtins.hpp

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText : © 2025 TU Wien <vadl@tuwien.ac.at>
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#ifndef VADL_BUILTINS_HPP
+#define VADL_BUILTINS_HPP
+
+#include <string>
+
+/*-----------------------------------------------------------------------
+ *    concat_str(a: String<N>, b: String<M>) -> String<X>
+ *---------------------------------------------------------------------*/
+static inline std::string VADL_concat_str(std::string a, std::string b) {
```

Comment:
major: Why is it here and not in `vadl/main/resources/templates/common/vadl-builtins.h`?